flocker/node/agents/ebs.py

```diff
@@ -875,6 +875,7 @@ def _next_device(self, instance_id, volumes, devices_in_use):
         IN_USE_DEVICES(devices=sorted_devices).write()
 
         for suffix in b"fghijklmonp":
+            # check for /dev/xvdX here too.
```

Comment:
Well... maybe? That might be the issue, but... we already presume that the device names are *always* mapped. So perhaps a better way to describing this issue is that this method mixes local devices with untranslated EC2-reported devices. Which is definitely a bug, it seems like the retry logic is wrong somehow since "ignored devices" is never actually useful.

But... we're getting all devices from EC2. Just because a third party attached one doesn't mean this should fail. It'll only fail if a third party added it recently enough that we hit eventual consistency issues (or in between list and attach). Is that what actually happened in the case you saw?

Since I believe we're fundamentally always assuming a mapping from EC2-reported `/dev/sd*` to local `/dev/xvd*`, perhaps the correct answer is to just look at local devices, filter to `/dev/xvd*` and only use those?